rpc/rpcutils/clients.go

```diff
@@ -79,7 +81,28 @@ func (rc *reconnectingClient) Call(serviceMethod string, args interface{}, reply
 		//get read lock again
 		rc.RLock()
 	}
-	err = rpcClient.Call(serviceMethod, args, reply)
+	if timeout == 0 {
+		timeout = 3153600000 // One hundred years in seconds.
+	}
+	c := make(chan error, 1)
+	go func() {
+		c <- rpcClient.Call(serviceMethod, args, reply)
+	}()
+	start := time.Now()
+Loop:
+	for {
+		select {
+		case err = <-c:
+			break Loop
+		case <-time.After(time.Duration(timeout) * time.Second):
+			err = fmt.Errorf("RPC call to %s timed out after %d seconds.", serviceMethod, timeout)
+			break Loop
+		case <-time.After(10 * time.Second):
+			if timewarn {
+				glog.Warningf("RPC call to %s has taken more than %ds.", serviceMethod, int(time.Since(start).Seconds()))
```

Comment:
The log message here should maybe have an identifier on it so that interleaved log messages for the same RPC call can be differentiated between